transformations/speech_disfluency_perturbation/transformation.py

```diff
@@ -0,0 +1,116 @@
+import random
+from typing import List, Optional
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+class SpeechDisfluencyPerturbation(SentenceOperation):
+    tasks: List[TaskType] = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+    ]
+    languages: List[str] = ["en"]
+    keywords = [
+        "lexical",
+        "noise",
+        "rule-based",
+        "highly-meaning-preserving",
+        "high-precision",
+        "high-coverage",
+        "high-generations",
+        "social-reasoning",
+    ]
+
+    filler_words: List[str] = ["um", "uh", "erm", "ah", "er"]
+    insert_prob: float
+
+    def __init__(
+        self,
+        seed: int = 0,
+        max_outputs: int = 1,
+        insert_prob: float = 0.2,
+        filler_words: Optional[List[str]] = None,
+    ) -> None:
+        super().__init__(seed, max_outputs=max_outputs)
+        self.insert_prob = insert_prob
+        if filler_words is not None:
+            self.filler_words = filler_words
+
+    def generate(self, sentence: str) -> List[str]:
+        random.seed(self.seed)
+        perturbed_texts = []
+
+        for _ in range(self.max_outputs):
+            tokens = sentence.split()
+
+            # In the case of very short inputs, allow for the addition of
+            # tokens before/after the first/last token respectively.
+            short_input = len(tokens) <= 1
+            if short_input:
+                tokens = [None] + tokens + [None]
+
+            choices = []
+            while not choices:
+                for i in range(1, len(tokens)):
+                    if random.random() < self.insert_prob:
+                        choices.append(i)
+
+            choices = choices[::-1]
+            for i in choices:
+                tokens.insert(i, random.choice(self.filler_words))
+
+            if short_input:
+                tokens = tokens[1:-1]
+
+            perturbed_texts.append(" ".join(tokens))
```

Comment:
You don't handle capitalization changes due to introduced words (which is nontrivial; you'd need to figure out what is a proper noun). I don't think it's that big of a deal, since this transformation is more oriented towards transcribed speech, but you should at least mention that in the "Limitations" section.

Reply:
I've updated the README to reflect the limitation. I can see problems with multi-sentence input (e.g., "Hello. How are you?" -> "Hello. uh How are you?"), but I don't quite follow why proper nouns are relevant ("His name is Bob Barker now" -> "His name is uh Bob Barker now"/"His name is Bob uh Barker now"/"His name is Bob Barker uh now")

Reply:
An example of the phenomenon I was thinking of: "I see why you like it here. Ahmed's house is right around the corner. The books are under the table" -> "I see why you like it here. Um **Ahmed's** house is right around the corner. Uh **The** books are under the table.". Notice that "Ahmed" should stay capitalized, while "The" should not.

Sorry for not describing it well enough/providing an example earlier!

Reply:
No worries; all makes sense now